src/modules/status-list/status-list.service.ts

```diff
@@ -0,0 +1,139 @@
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { verifyCredential } from 'didkit-wasm-node';
+import {
+  CredentialWithStatusDto,
+  StatusList2021EntryDto,
+} from './dtos/credential-status.dto';
+import { STATUS_LIST_MODULE_PATH } from './status-list.const';
+import { CredentialDto } from './dtos/credential.dto';
+import { VerifiableCredentialDto } from './dtos/verifiable-credential.dto';
+import { StatusListCredentialDto } from './dtos/status-list-credential.dto';
+import { StatusListVerifiableCredentialDto } from './dtos/status-list-verifiable-credential.dto';
+import { StatusListCredential } from './entities/status-list-credential.entity';
+
+@Injectable()
+export class StatusListService {
+  constructor(
+    private readonly configService: ConfigService,
+    @InjectRepository(StatusListCredential)
+    private readonly statusListRepository: Repository<StatusListCredential>
+  ) {}
+
+  async createEntry(
+    credential: CredentialDto
+  ): Promise<CredentialWithStatusDto> {
+    if (await this.fetchStatusList(credential.id)) {
+      throw new ConflictException('Credential already revoked');
+    }
+
+    const statusListCredential = new URL(
+      `${STATUS_LIST_MODULE_PATH}/${credential.id}`,
+      this.configService.get('STATUS_LIST_DOMAIN')
+    );
+
+    // https://w3c-ccg.github.io/vc-status-list-2021/#statuslist2021entrya
+    const credentialStatus: StatusList2021EntryDto = {
+      type: 'StatusList2021Entry',
+      statusPurpose: 'revocation',
+      statusListIndex: '1', // because we are using one-to-one mapping
+      statusListCredential: statusListCredential.href,
+    };
+
+    return { ...credential, credentialStatus };
+  }
+
+  async createStatusListCredential(
+    credential: VerifiableCredentialDto,
+    currentUser: string
+  ): Promise<StatusListCredentialDto> {
+    const statusList = await this.statusListRepository.findOne({
+      where: {
+        credentialId: credential.id,
+      },
+    });
+
+    if (statusList) {
+      throw new ConflictException('Credential already revoked');
+    }
+
+    const list = await StatusList.createList({ length: 1 });
+    list.setStatus(+credential.credentialStatus.statusListIndex, true);
+    const encodedList = await list.encode();
+
+    // https://w3c-ccg.github.io/vc-status-list-2021/#statuslist2021credential
+    const statusListCredential = {
+      '@context': [
+        'https://www.w3.org/2018/credentials/v1',
+        'https://w3id.org/vc/status-list/2021/v1',
+      ],
+      id: credential.credentialStatus.statusListCredential,
+      type: ['VerifiableCredential', 'StatusList2021Credential'],
+      issuer: currentUser,
+      issued: new Date().toISOString(),
+      credentialSubject: {
+        id: credential.id,
+        type: 'StatusList2021',
+        statusPurpose: 'revocation',
+        encodedList,
+      },
+    };
+
+    return statusListCredential;
+  }
+
+  async addSignedStatusList(
```

Comment:
Could you add description to this method?

Reply:
Yes, I will add it